### PR TITLE
Cleaning up the pod utility functions into their own pkg

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

we have too many _util files, trying to clean that up under the util/ pkg

/kind cleanup

#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/controller/pod_controller.go [hunk 1]
```diff
@@ -211,6 +216,178 @@ func (r *PodReconciler) topologyValueFromPod(ctx context.Context, pod *corev1.Po
 	return topology, nil
 }
 
+// constructLeaderStatefulSetApplyConfiguration constructs the apply configuration for the leader StatefulSet
+func constructLeaderStatefulSetApplyConfiguration(lws *leaderworkerset.LeaderWorkerSet) (*appsapplyv1.StatefulSetApplyConfiguration, error) {
```
Comment (ahg-g) [inline pkg/controller/pod_controller.go:220]: this is simply moving the function from the now deleted controller_util.go file

**Review by kerthcet — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**